frappe/deprecation_dumpster.py

```diff
@@ -200,6 +201,50 @@ def deprecation_warning(marked: str, graduation: str, msg: str):
 
 
 ### Party starts here
+
+if typing.TYPE_CHECKING:
+	from werkzeug.local import Local
+
+
+def get_local_with_deprecations() -> "Local":
+	from werkzeug.local import Local
+
+	class DeprecatedLocalAttribute:
+		def __init__(self, name, warning):
+			self.name = name
+			self.warning = warning
+
+		def __get__(self, obj, type=None):
+			self.warning()
+			return obj.__getattr__(self.name)
+
+		def __set__(self, obj, value):
+			return obj.__setattr__(self.name, value)
+
+		def __delete__(self, obj):
+			return obj.__delattr__(self.name)
+
+	class LocalWithDeprecations(Local):
+		sites_path = DeprecatedLocalAttribute(
+			"sites_path",
+			lambda: deprecation_warning(
+				"2024-12-06",
+				"v17",
+				"'local.sites_path' will be deprecated: use 'frappe.bench.sites.path instead'",
```

Comment:
There is no need to do all this. We can just drop this without warning. 

It's sufficiently "internal" to Frappe framework.

Reply:
Indeed, I thought so. But I'd like to keep it as a stepping store for my own sake during the refactoring stream (this is meant to be only the first PR). I have an idea though: we could set the graduation to v15, this way it provokes hard errors.

Once the refactoring stream is over and no more errors left, we can remove it immediately.

Reply:
It introduces a lot more code than previous version, which honestly is simpler than this at present.

Reply:
Also `bencher.py` is weird naming.

Reply:
> It introduces a lot more code than previous version, which honestly is simpler than this at present.

I'm just a bit afraid at breaking something on the way, I'd prefer to keep it as a temporary band-aid, just to be on the safe side and honor the stability requirement. But if you feel strongly about, we can also drop it right away.

> Also bencher.py is weird naming.

`bench.py` whould have been ideal, but I needed the namespace in `frappe.bench`, so I thought the next best option the file which is responsible for bench layout, i.e. the "maker of bench-compatibility", short "bencher". Idk, I'm not particularly attached to it, but prefer it to `_bench.py` which would slgnal something "internal", which it explicitly is not.

Reply:
Has anyone else signed on to this change? 

@akhilnarang or @sumitbhanushali ? 

I'd highly advise against self-merging opinionated changes. I'll be resuming work soon-ish. I am glad to provide an opinion wherever needed. I just don't want to see this codebase become a hodge-podge of 2 different styles.

Reply:
E.g. What even is this? Environment variables that depend on class name? https://github.com/frappe/frappe/blob/3d71f594d8b1276777eee168dad4c5dec0148264/frappe/bencher.py#L37-L41

Reply:
> But if you feel strongly about

Or "sufficiently more confident than me", for that matter. :smile:

Reply:
It's well documented: https://github.com/frappe/erpnext/wiki/Naming-Guidelines 

It should at least resemble what it does. `bench_resolver` maybe? (I haven't read entire code)

Reply:
> I'd highly advise against self-merging opinionated changes.

Not gonna happen. :smile: This had been coordinated at large with @akhilnarang on https://github.com/frappe/frappe/pull/28158 over TG.

Reply:
All actionable feedback has been addressed in followups to #28158